scripts/consensus_merge.py

```diff
@@ -406,21 +407,36 @@ def get_dp_consensus(dp_list):
     """
     return int(np.mean([int(f) for f in dp_list if f != '.']))
 
-def get_mapq(cram, chrom, pos):
+def get_mapq(cram_path, chrom, pos, reference=None):
     """ Get RMS mapping quality and number of MAPQ=0 reads at locus
 
         Args:
-            cram (pysam.AlignmentFile): open in 'r' mode
+            cram_path (str): path to CRAM or BAM file
             chrom (str): locus chromosome name
             pos (int): 1-based locus coordinate
+            reference: path to reference FASTA, required for CRAM
+                (default None)
+
+        Raises:
+            IOError if CRAM provided without reference
+                    if cram_path has neither cram nor bam extension
 
         Returns:
             tuple of ints RMS mapq and # of mapq reads at locus
     """
+    if cram_path.endswith('cram'):
+        if not reference:
+            raise IOError('Must provide reference with CRAM')
+        cram = pysam.AlignmentFile(cram_path, 'rc', reference_filename=reference)
+    elif cram_path.endswith('bam'):
+        cram = pysam.AlignmentFile(cram_path, 'rb')
+    else:
+        raise IOError('File provided to "cram" argument must have .cram or .bam extension')
```

Comment:
So would the old method of giving this function a `pysam.AlignmentFile` not work/work slower with the multiprocessing?

Reply:
With `multiprocessing` you can only pass around objects that can be serialized with `pickle`. This is often an annoying limitation as random objects like AlignmentFiles turn out not to pickle. And it turns out there's no profound reason for it! [Some guy](https://pypi.org/user/mmckerns/) produced a module called `dill` that _can_ serialize arbitrary python objects, and he made a fork of `multiprocessing` on top of that overcomes these limitations, which I have used before. It's not exactly a standard-library module, though, and I decided not to use it here on the theory that there wouldn't be any major speed difference, but I didn't actually try to find out.

Reply:
Cool! Thanks for the lesson. I've never used multiprocessing before so this is good knowledge to have.